feconf.py

```diff
@@ -37,9 +37,9 @@
 # When DEV_MODE is true check that we are running in development environment.
 # The SERVER_SOFTWARE environment variable does not exist in Travis, hence the
 # need for an explicit check.
-if (constants.DEV_MODE and os.getenv('SERVER_SOFTWARE') and
-        not os.getenv('SERVER_SOFTWARE', default='').startswith('Development')):
-    raise Exception('DEV_MODE can\'t be true on production.')
+assert not (constants.DEV_MODE and os.getenv('SERVER_SOFTWARE') and
```

Comment:
Asserts are apparently removed when compilation is optimized, so don't use an assert here: https://stackoverflow.com/a/1838411

You should be able to test this by swapping accordingly.

Reply:
Yes I can test it by swapping if for example these lines were in a function. Then, I could do something like:
```python
with self.swap():
    feconf.func()
```
But since they are in the outermost level of a file, do you know how I can call this block? I tried importing feconf but the exception is not being raised. I also tried to call it directly by subprocess but still the exception is not getting raised.

Reply:
This is a complex if statement. Which specific part are you having trouble swapping out?

Reply:
No I am not having trouble with swapping anything, I am just asking how can I call this block of code? For example, if the lines (40-42) were in a function `func` I can call feconf.func() but this block is at the outermost indent. I tried simple importing and running subprocess but the exception is not triggering

Reply:
The entirety of feconf.py is run once it is imported. If you put a "raise Exception" statement on line 39, say, I think you will see it.

To call this block you will need to set things up so that the conditions to make the "if" statement true hold.

Reply:
Done